File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let map = new Map()
        for(let i = 0; i<nums.length; i++){
            let current = target-nums[i]
            if(map.has(current)) return [map.get(current), i]
            map.set(nums[i],i);
        }
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int n = numbers.length;
        int j = n - 1;
        for (int i = 0; i < n;) {
            if (numbers[i] + numbers[j] == target)
                return new int[] {i + 1, j + 1};
            else if (numbers[i] + numbers[j] > target)
                j--;
            else if (numbers[i] + numbers[j] < target)
                i++;
        }
        return new int[] {};
    }
}
